src/Runtime/Distributions/SequenceDistribution.cs

```diff
@@ -1106,15 +1106,7 @@ public void SetTo(TThis that)
             }
 
             this.point = that.point;
-            if (this.point == null)
-            {
-                this.SetWeightFunction(that.sequenceToWeight);
-            }
-            else
-            {
-                this.sequenceToWeight = null;
-            }
-
+            this.sequenceToWeight = that.sequenceToWeight;
```

Comment:
Why don't we need to clone here?

Reply:
That was a bug! Fixed it before seeing your comment